### PR TITLE
Translation support

### PR DESCRIPTION
**Pull request by JulianGro**
This PR adds translation support to the docs.
It depends on #48 and #59 as it is based on them.

Here is a list of what it changes/adds:
- GitHub Actions
  - Build and deploy all languages with every commit to master
  - Build and deploy english for every PR (updates with every commit)
  - Comment on PR where PR builds are available
- add some more info to the README.md
- add MAINTAINER_INFO.md file to keep track of bits and bobs needed to maintain the translation system
- add language selector (with static URLs) to the bottom left
- add available languages and translation status to home.rst
- change image styling to allow for the translation status to be displayed correctly, this moved all images from the middle to the left, which looked better to me anyways
- fix wrong markdown setup in conf.py (for some reason sphinx 2.4.4 did not care, but 3.x.x breaks completely without this fix)
- disable html_copy_source since it only wastes space and slows down github actions
- add japanese translation files

I also added automatic language file updates to the weblate server via git hooks.

Currently GitHub Actions is configured to work with my branch and server.
Before merging we will have to configure GHA to work with the master branch and Kalila's server. We also need to configure the repo and Kalila's server for this.

Big thanks to Thoys for helping out whenever I got stuck.

### CODE REVIEW

**Comment by JulianGro** [comment]
@thoys would be cool is you could CR the github actions.

File: .github/workflows/master_build.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        cd docs
+        make html
+        mkdir -p build-archives
+        cd build/html/
+        tar -czvf ../../build-archives/build-en.tar.gz *
+    - name: Upload documentation
+      uses: JulianGro/SFTP-Deploy-Action@v1.4
+      with:
+        username: 'gha'
+        server: '94.130.177.235'
```
Comment (thoys) [inline .github/workflows/master_build.yml:41]: I see this ip address repeated over the entire file, perhaps consider setting it as a variable that can be re-used through the script. same for user name.  In case they change you'd only have to change those it will only be in one place.

File: .github/workflows/master_build.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+name: Master CI Build
+
+on:
+  push:
+    branches:
+#      - master
```
Comment (thoys) [inline .github/workflows/master_build.yml:6]: you'd have to uncomment this to make it work on the master branch

File: .github/workflows/pr_build.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        username: 'gha'
+        server: '94.130.177.235'
+        ssh_private_key: ${{ secrets.SSH_PRIVATE_KEY }}
+        local_path: 'docs/build-archives/*'
+        remote_path: '/var/www/html/gha/PR${{ github.event.number }}/'
+        args: '-o ConnectTimeout=5'
+    - name: Deploy english documentation
+      shell: bash
+      run: |
+        printf "%s" "${{ secrets.SSH_PRIVATE_KEY }}" >private_key.pem
```
Comment (thoys) [inline .github/workflows/pr_build.yml:47]: A space after the `>` would probably make it a bit more readable
```suggestion
        printf "%s" "${{ secrets.SSH_PRIVATE_KEY }}" > private_key.pem
```

File: .github/workflows/master_build.yml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        remote_path: '/var/www/html/gha/'
+        args: '-o ConnectTimeout=5'
+    - name: Deploy english documentation
+      shell: bash
+      run: |
+        printf "%s" "${{ secrets.SSH_PRIVATE_KEY }}" >private_key.pem
+        chmod 600 private_key.pem
+        ssh -o StrictHostKeyChecking=no -i private_key.pem gha@94.130.177.235 mkdir -p /var/www/html/gha/
+        ssh -o StrictHostKeyChecking=no -i private_key.pem gha@94.130.177.235 tar -xvzf /var/www/html/gha/build-en.tar.gz -C /var/www/html/gha/
+        ssh -o StrictHostKeyChecking=no -i private_key.pem gha@94.130.177.235 rm /var/www/html/gha/build-en.tar.gz
```
Comment (thoys) [inline .github/workflows/master_build.yml:53]: You're connecting 3 times to the ssh here. How about running the ssh command once while sending multiple commands perhaps using here docs https://bash.cyberciti.biz/guide/Here_documents or more ssh specific help page https://www.cyberciti.biz/faq/linux-unix-osx-bsd-ssh-run-command-on-remote-machine-server/ , which would make it look like
```
ssh -o StrictHostKeyChecking=no -i private_key.pem gha@94.130.177.235 << EOF
    mkdir -p /var/www/html/gha/
    tar -xvzf /var/www/html/gha/build-en.tar.gz -C /var/www/html/gha/
    rm /var/www/html/gha/build-en.tar.gz
EOF
```

File: .github/workflows/pr_build_info.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: mshick/add-pr-comment@v1
+      with:
+        message: |
+          **This PR will be available at** http://94.130.177.235/gha/PR${{ github.event.number }}/ **once building has completed.**
+          Every new commit will trigger a rebuild. Look at the "Checks" to see progress.
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        repo-token-user-login: 'Buildsystem' # The user.login for temporary GitHub tokens
+        allow-repeats: false # This is the default
```
Comment (thoys) [inline .github/workflows/pr_build_info.yml:18]: Add missing line ending, preferred on unix based systems

File: .github/workflows/master_build.yml (hunk 1 above)
Comment (JulianGro) [reply .github/workflows/master_build.yml:41]: From what I can see online, there isn't actually normal variables.
I could set environment variables, but those do not persist between runners, which is pretty much the whole point.

File: .github/workflows/master_build.yml (hunk 2 above)
Comment (JulianGro) [reply .github/workflows/master_build.yml:6]: This is one of the things that need to be changed before actually merging.
I am just leaving all the "old" settings in for now, so weblate can still be used on this branch.

File: .github/workflows/master_build.yml (hunk 4 above)
Comment (JulianGro) [reply .github/workflows/master_build.yml:53]: As I don't think those commands are run as a bash script, but rather just as bash commands, I changed them to one-liners.

File: .github/workflows/pr_build_info.yml (hunk 5 above)
Comment (JulianGro) [reply .github/workflows/pr_build_info.yml]: Changed the setting in my IDE. Should have done it.

File: docs/source/home.rst [hunk 6]
```diff
@@ -4,7 +4,20 @@ Home
 
 Vircadia is an open-source software where you can create and share virtual reality (VR) experiences. You can create and host your own VR world, explore other worlds, meet and connect with other users, attend or host live VR events and much more.
 
-The Vircadia metaverse provides built-in social features, including avatar interactions, spatialized audio and interactive physics. Additionally, you have the ability to import any 3D object into your virtual environment. No matter where you go in Vircadia, you will always be able to interact with your environment, engage with your friends, and listen to conversations just like you would in real life. 
+The Vircadia metaverse provides built-in social features, including avatar interactions, spatialized audio and interactive physics. Additionally, you have the ability to import any 3D object into your virtual environment. No matter where you go in Vircadia, you will always be able to interact with your environment, engage with your friends, and listen to conversations just like you would in real life.
+
+This documentation is authored in `english <https://docs.vircadia.dev/>`_.
```
Comment (ctrlaltdavid) [inline docs/source/home.rst:9]:
```suggestion
This documentation is authored in `English <https://docs.vircadia.dev/>`_.
```

File: docs/source/_static/css/custom.css [hunk 7]
```diff
@@ -29,9 +29,9 @@ body {
 }
 
 img {
-    display: block;
+#    display: block;
     text-align: center;
-    margin: 2rem auto;
+#    margin: 2rem auto;
```
Comment (ctrlaltdavid) [inline docs/source/_static/css/custom.css:34]: Why this change? Now all images are left-aligned on the page instead of centred.

Comment (JulianGro) [reply docs/source/_static/css/custom.css:34]: Yes. This is to allow the translation status to be displayed on the main page: http://94.130.177.235/gha/
When looking around affected images I came to the conclusion that this looked better anyways.
I have unfortunately not been able to get the translation status properly aligned without this change.

Comment (ctrlaltdavid) [reply docs/source/_static/css/custom.css:34]: Let's revert these changes and give the  elements a CSS class of "inline". ... 

Add the following CSS rule underneath the above reverted rule:
```
img.inline {
    display: inline-block;
    margin: 0;
}
```

And specify the class in home.rst per my suggested change there.

This will style the translation images as you want while leaving the remainder of the images in the docs centred.

File: docs/source/home.rst [hunk 8]
```diff
@@ -4,7 +4,20 @@ Home
...
+The following translations are available:
+
+* `Deutsch <https://docs.vircadia.dev/de/>`_ |de_trans|
+* `日本語 <https://docs.vircadia.dev/jp/>`_ |jp_trans|
+
+.. |de_trans| image:: https://weblate.vircadia.dev/widgets/vircadia-documentation/de/svg-badge.svg
+              :target: https://weblate.vircadia.dev/engage/vircadia-documentation/de/?utm_source=widget
+
+.. |jp_trans| image:: https://weblate.vircadia.dev/widgets/vircadia-documentation/ja/svg-badge.svg
+              :target: https://weblate.vircadia.dev/engage/vircadia-documentation/ja/?utm_source=widget
```
Comment (ctrlaltdavid) [inline docs/source/home.rst:20]:
```suggestion
.. |de_trans| image:: https://weblate.vircadia.dev/widgets/vircadia-documentation/de/svg-badge.svg
              :class: inline
              :target: https://weblate.vircadia.dev/engage/vircadia-documentation/de/?utm_source=widget

.. |jp_trans| image:: https://weblate.vircadia.dev/widgets/vircadia-documentation/ja/svg-badge.svg
              :class: inline
              :target: https://weblate.vircadia.dev/engage/vircadia-documentation/ja/?utm_source=widget
```
Makes  tags CSS class "inline".

Comment (JulianGro) [reply docs/source/home.rst]: Very nice. I had given up on trying to get them working seperately.
You can see it working on the GHA build (might have to clear browser cache).
